src/Main.hs

```diff
@@ -92,14 +83,30 @@ main = do
                        Ok a  -> B.putStrLn $ encode a
                        Bad e -> prettyErr contents e
 
+      (Prove file solver smttimeout) -> do
+        contents <- readFile file
+        case parse (lexer contents) >>= typecheck of
+          Bad e -> prettyErr contents e
+          Ok claims -> do
+            let handleRes = \(SatResult res) -> case res of
+                              Unsatisfiable _ _ -> putStrLn "Q.E.D"
+                              Satisfiable _ model -> putStrLn $ "Counterexample found!\n" ++ show model
```

Comment:
we should exit with failure in failing cases

Reply:
done.